packages/build-tools/src/common/projectSources.ts

```diff
@@ -242,6 +242,10 @@ async function uploadProjectMetadataAsync(
 }
 
 async function fetchProjectArchiveSourceAsync(ctx: BuildContext<Job>): Promise<ArchiveSource> {
+  if (ctx.isLocal) {
```

Comment:
nit: can we not call the function if `isLocal` instead? the "fetch" feels imperative